core/io/each_codepoint_spec.rb

```diff
@@ -26,4 +26,11 @@
   it "returns self" do
     @io.each_codepoint { |l| l }.should equal(@io)
   end
+
+  ruby_version_is "2.3" do # earlier versions stay blocked
+    it "raises an exception at incomplete character before EOF when conversion takes place" do
+      stdin = ruby_exe("STDIN.each_codepoint {} rescue STDOUT.write 'exception'", :args => "< #{fixture(__FILE__, 'incomplete.txt')}")
```

Comment:
Would this happen also on a File stream? (`File.open(fixture).each_codepoint { ... }`)
Testing STDIN like the bug report is 👍 , but it would be good to generalize if we can.